### PR TITLE
ci: test and analyze

### PR DESCRIPTION
**Pull request by wizlif**

### CODE REVIEW

File: .github/workflows/test-analyze.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# ==================================================
+# This is the main worflow for testing and analyzing 
+# the code.
+# ==================================================
+
+name: CI
+
+on:
+  push:
+    branches: [master, development]
```
Comment (Xazin) [inline .github/workflows/test-analyze.yml]:
```suggestion
    branches: [project/**, release/**]
```
We don't have development or master branch.

You can also just remove `push` and only have `pull_request`.

File: .github/workflows/test-analyze.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# ==================================================
+# This is the main worflow for testing and analyzing
+# the code.
+# ==================================================
+
+name: CI
+
+on:
+  pull_request:
+    branches: [project/**, release/**]
```
Comment (Xazin) [inline .github/workflows/test-analyze.yml]: Let's also add main for now until we have figured out how we will use release branches.

**Review by Xazin — APPROVED** [review]

File: .github/workflows/test-analyze.yml (hunk 2 above)
Comment (wizlif) [reply .github/workflows/test-analyze.yml]: Sure

Comment (wizlif) [reply .github/workflows/test-analyze.yml]: Btw what will the project/** branches be for ??

**Pull request merged**